packages/vm/tests/BlockchainTestsRunner.ts

```diff
@@ -73,6 +73,10 @@ export default async function runBlockchainTest(options: any, testData: any, t:
     common,
   })
 
+  // Need to await the init promise: in some tests, we do not run the iterator (which awaits the initPromise)
+  // If the initPromise does not finish, the `rawHead` of `blockchain.meta()` is still `undefined`.
+  await blockchain.initPromise
```

Comment:
Do you think there is any chance users of the blockchain library may run into this?

If so then as you previously suggested we can make an async public factory method to generate the blockchain and then make the constructor private to reduce the chance of this happening. For example libp2p uses `.create()` as an async method (although they don't restrict the use of the constructor, in their case if you supply a PeerId then it doesn't need to be async and the constructor can be used)